worlds/pokemon_crystal/__init__.py

```diff
@@ -0,0 +1,333 @@
+        md5s = ["9f2922b235a5eeb78d65594e82ef5dde"]
+
+    class RomStart(str):
+        """
+        Set this to false to never autostart a rom (such as after patching)
+        True for operating system default program
+        Alternatively, a path to a program to open the .gb file with
+        """
+
+    rom_file: RomFile = RomFile(RomFile.copy_to)
+    rom_start: Union[RomStart, bool] = True
+
+
+class PokemonCrystalWebWorld(WebWorld):
+    tutorials = [Tutorial(
+        "Multiworld Setup Guide",
+        "A guide to playing Pokemon Crystal with Archipelago.",
+        "English",
+        "setup_en.md",
+        "setup/en",
+        ["AliceMousie"]
+    )]
+
+
+class PokemonCrystalWorld(World):
+    """Pokémon Crystal is the culmination of the Generation I and II Pokémon games.
+    Explore the Johto and Kanto regions, become the Pokémon League Champion, and
+    defeat the elusive Red at the peak of Mt. Silver!"""
+    game = "Pokemon Crystal"
+
+    topology_present = True
+    web = PokemonCrystalWebWorld()
+
+    settings_key = "pokemon_crystal_settings"
+    settings: ClassVar[PokemonCrystalSettings]
+
+    options_dataclass = PokemonCrystalOptions
+    options: PokemonCrystalOptions
+
+    data_version = 0
+    required_client_version = (0, 4, 4)
+
+    item_name_to_id = create_item_label_to_code_map()
+    location_name_to_id = create_location_label_to_id_map()
+    item_name_groups = ITEM_GROUPS  # item_groups
+
+    free_fly_location: int
+    map_card_fly_location: int
+    generated_pokemon: Dict[str, PokemonData]
+    generated_starters: Tuple[List[str], List[str], List[str]]
+    generated_starter_helditems: Tuple[str, str, str]
+    generated_trainers: Dict[str, TrainerData]
+    generated_palettes: Dict[str, List[int]]
+    generated_phone_traps: List[PhoneScript]
+    generated_phone_indices: List[int]
+    generated_misc: MiscData
+    generated_tms: Dict[str, TMHMData]
+    generated_wild: WildData
+    generated_music: List[int]
+    generated_wooper: str
+    generated_static: Dict[str, StaticPokemon]
+
+    def generate_early(self) -> None:
+        if self.options.early_fly:
+            self.multiworld.local_early_items[self.player]["HM02 Fly"] = 1
+            if (self.options.hm_badge_requirements.value != HMBadgeRequirements.option_no_badges
+                    and self.options.randomize_badges == RandomizeBadges.option_completely_random):
+                self.multiworld.local_early_items[self.player]["Storm Badge"] = 1
+
+        if self.options.johto_only:
+            if self.options.goal == Goal.option_red and self.options.johto_only == JohtoOnly.option_on:
+                self.options.goal.value = Goal.option_elite_four
+                logging.warning(
+                    "Pokemon Crystal: Red goal is incompatible with Johto Only "
+                    "without Silver Cave. Changing goal to Elite Four for player %s.",
+                    self.multiworld.get_player_name(self.player))
```

Comment:
I thought about this change too, but it makes random yamls *extremely* likely to error and the warning is printed, which should be informative

Reply:
Hmm. If it were me, I think I'd just tell my players to deal with it and make a trigger because I can't know which behavior they expected. People won't see the generation log if they use the website, so won't see warnings. And I'm sure even when someone generates locally and there are warnings, they often aren't relayed from host to player.

It's a pick your poison sort of situation. I won't say either is definitively the best choice.